src-tauri/src/state/reducers.rs

```diff
@@ -0,0 +1,111 @@
+use crate::state::actions::Action;
+use crate::state::state::{AppState, Profile, StateStatus};
+use tracing::{warn, info};
+
+/// Sets the locale to the given value. If the locale is not supported yet, the current locale will stay unchanged.
+pub fn set_locale(state: &AppState, action: Action) -> anyhow::Result<()> {
+    let payload = action.payload.unwrap();
+    let locale = payload["locale"].as_str().expect("unable to read locale from json payload");
```

Comment:
Using `expect()` will also result in a panic when there's invalid input. You can use `ok_or()` here as well.
```suggestion
    let locale = payload["locale"]
        .as_str()
        .ok_or(anyhow::anyhow!("unable to read locale from json payload"))?;
```